dev/bots/test/analyze_test.dart

```diff
@@ -208,4 +209,116 @@ void main() {
     expect(result, contains(':20'));
     expect(result, contains(':21'));
   });
+
+  test('analyze.dart - clampDouble', () async {
+    final String result = await capture(() => runVerifiersInResolvedDirectory(
+      testRootPath,
+      <ResolvedUnitVerifier>[verifyNoDoubleClamp],
+    ), shouldHaveErrors: true);
+    final String lines = <String>[
+        '║ packages/flutter/lib/bar.dart:37: input.clamp(0.0, 2)',
+        '║ packages/flutter/lib/bar.dart:38: input.toDouble().clamp(0, 2)',
+        '║ packages/flutter/lib/bar.dart:42: nullableInt?.clamp(0, 2.0)',
+        '║ packages/flutter/lib/bar.dart:43: nullableDouble?.clamp(0, 2)',
+        '║ packages/flutter/lib/bar.dart:48: nullableInt?.clamp',
+        '║ packages/flutter/lib/bar.dart:50: nullableDouble?.clamp',
+      ]
+      .map((String line) => line.replaceAll('/', Platform.isWindows ? r'\' : '/'))
+      .join('\n');
+    expect(result,
+      '╔═╡ERROR╞═══════════════════════════════════════════════════════════════════════\n'
+      '$lines\n'
+      '║ \n'
+      '║ For performance reasons, we use a custom "clampDouble" function instead of using "double.clamp".\n'
+      '║ For non-double uses of "clamp", use "// ignore_clamp_double_lint" on the line to silence this message.\n'
```

Comment:
Why is this ignore still necessary? I thought the new analysis would be smart enough to only flag actual `double.clamp` usage?

Reply:
Yeah I can remove that for now.